Improvements to the Repository Settings

madoar opened the pull request:
This PR tries to improve some points from #667.
More detailed it adds the following features to the repositories list in the repositories settings:

- When viewing the repositories list in the repositories settings you can now see the priorities of the repositories, via a number in front of the repository
- The order of the repositories in the repository list is now changeable via drag and drop. The current behavior is, that directly after the order was changed the new repository order is saved to disk, maybe we want to change that and add a "Save"-button.

![grafik](https://cloud.githubusercontent.com/assets/18488086/24115225/f377653c-0da2-11e7-9634-dae7b08bb9f1.png)

A still open problem is, that this PR breaks the text-color of the `ListCell`-content. I guess that this is because this PR changes the structure of the `ListCell`-content, which was previously simply a string. Now the text content is wrapped inside Labels, which are inside a GridPane, which is added to the ListCell as its graphic.

![grafik](https://cloud.githubusercontent.com/assets/18488086/24116020/959bab78-0da5-11e7-8632-435509a73b71.png)

qparis commented on phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java:
```diff
@@ -0,0 +1,121 @@
+/**
+ * 
```
Please remove this

qparis commented on phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java:
```diff
@@ -0,0 +1,121 @@
+	/**
+	 * What should be done after this cell has been dragged
+	 */
+	private Runnable onDragDone;
```
Can this be final

qparis commented on phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java:
```diff
@@ -0,0 +1,121 @@
+public class DragableRepositoryListCell extends ListCell<String> implements ChangeListener<Number> {
+
+	/**
+	 * What should be done after this cell has been dragged
```
This comment is not needed, however, the corresponding javadoc can be written on the constructor

plata commented:
Why do we have numbers and an order? Isn't this duplicate? Also, I think that it is not clear what the meaning of the numbers is.
I would prefer to have just the order and add an explanatory text describing that you can change the priority by drag and drop.

madoar commented:
The problem currently is, that the order of the List isn't visible to the user. What is the repository with the highest and the lowest priority?
To visualize this I've added a number that everyone can understand. I agree with you that we should add a short legend for this.

madoar commented:
I've just fixed the css problem, and removed the center alignment of the priority.

About the priority information in general:
I'm not sure how to handle them in the final version. But I think we need to somehow show the user what priority which repository has. Therefore I would suggest merging this now and postponing the final decision how to do this to a later time.

plata commented:
I would like to have the legend in this PR already. Should not be a big deal.

madoar commented:
I was just too lazy to do that^^
![grafik](https://cloud.githubusercontent.com/assets/18488086/24212643/4efd211a-0f30-11e7-841f-bb3dc6425a58.png)

![grafik](https://cloud.githubusercontent.com/assets/18488086/24212660/5a03ea44-0f30-11e7-9948-5222bcfdfb19.png)

madoar commented:
Just as a note: I think somewhere between the file reading and the loading of the repositories the list gets reversed. We should check where that happens (and if it really happens...), because I don't think we should reverse the list, this only makes working with the repository list more complex.
